### PR TITLE
docs: update style guide

### PR DESCRIPTION
**Pull request by erickzhao**
#### Description of Change

This PR updates http://electronjs.org/docs/styleguide.

Notable changes:
* Clarified verbiage around titles to use `heading` (the CommonMark spec term).
* Re-organized API section to be clearer.
* Removed rule for 80-column lines. I like having shorter lines, but this would require a lot of manual line breaking across our documentation, since we haven't adhered to this rule very well.
* Added a TODO for the markdown linter to match these markdown rules.

cc @molant @zcbenz

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

File: docs/styleguide.md [hunk 1]
```diff
@@ -43,13 +42,12 @@ Using `Quick Start` as example:
 For API references, there are exceptions to this rule.
 
 ## Markdown rules
-
+<!--TODO(erickzhao): make sure this matches with the lint:markdownlint task-->
 * Use `sh` instead of `cmd` in code blocks (due to the syntax highlighter).
-* Lines should be wrapped at 80 columns.
```
Comment (malept) [inline docs/styleguide.md:48]: IMO, folks should prefer to wrap at 80 or 100 columns for readability, but it's not necessary to enforce.

File: docs/styleguide.md [hunk 2]
```diff
@@ -2,15 +2,14 @@
...
-* Chapters in the same page must have `##`-level titles.
-* Sub-chapters need to increase the number of `#` in the title according to
+* Chapters in the same page must have `##`-level headings.
+* Sub-chapters need to increase the number of `#` in the heading according to
   their nesting depth.
-* All words in the page's title must be capitalized, except for conjunctions
-  like "of" and "and" .
-* Only the first word of a chapter title must be capitalized.
+* The page's title must follow [APA title case][title-case]
+* All chapters must follow [APA sentence case][sentence-case]
```
Comment (malept) [inline docs/styleguide.md:12]: We should probably be consistent with periods at the end of bullet points.
```suggestion
* The page's title must follow [APA title case][title-case].
* All chapters must follow [APA sentence case][sentence-case].
```

There are other instances further down in the PR but I don't really have the time to annotate them all at the moment.

File: docs/styleguide.md [hunk 3]
```diff
@@ -226,9 +238,12 @@ The properties chapter must be in following form:
 ...
 ```
 
-The title can be `###` or `####`-levels depending on whether it is a property of
-a module or a class.
+The heading can be `###` or `####`-levels depending on whether the property
+belongs to a module or a class.
 
 ## Documentation Translations
```
Comment (malept) [inline docs/styleguide.md:244]: Should be `Document translations` per the title case documentation?

Comment (erickzhao) [reply docs/styleguide.md:244]: True!

File: docs/styleguide.md (hunk 1 above)
Comment (erickzhao) [reply docs/styleguide.md:48]: Sure, I'll add that nuance in.

**Review by zcbenz — APPROVED** [review]

**Pull request merged**